### PR TITLE
[#7327] Filesystem library now understands 4.2 permission strings (main)

### PR DESCRIPTION
**Pull request by korydraughn**
Will run tests soon.

### CODE REVIEW

File: lib/filesystem/src/filesystem.cpp [hunk 1]
```diff
@@ -104,18 +104,22 @@ namespace irods::experimental::filesystem::NAMESPACE_IMPL
...
             if (_perm == "create_metadata") { return perms::create_metadata; }
             if (_perm == "modify_metadata") { return perms::modify_metadata; }
             if (_perm == "delete_metadata") { return perms::delete_metadata; }
             if (_perm == "create_object")   { return perms::create_object; }
             if (_perm == "modify_object")   { return perms::write; }
+            if (_perm == "modify object")   { return perms::write; }
             if (_perm == "write")           { return perms::write; }
             if (_perm == "delete_object")   { return perms::delete_object; }
             if (_perm == "own")             { return perms::own; }
+            if (_perm == "null")            { return perms::null; }
```
Comment (trel) [inline lib/filesystem/src/filesystem.cpp:118]: consider putting 'null' at the 'top' of the list.

Comment (korydraughn) [reply lib/filesystem/src/filesystem.cpp]: Done

**Review by alanking — APPROVED** [review]
Add #

**Comment by korydraughn** [comment]
Added #

**Pull request merged**